src/network/networkd-address.c

```diff
@@ -485,10 +485,16 @@ static void log_address_debug(const Address *address, const char *str, const Lin
                                                     address->cinfo.ifa_valid * USEC_PER_SEC,
                                                     USEC_PER_SEC);
 
-                log_link_debug(link, "%s address: %s%s%s/%u (valid %s%s)",
+                if (address->cinfo.ifa_prefered != CACHE_INFO_INFINITY_LIFE_TIME)
+                        preferred_str = format_timespan(preferred_buf, FORMAT_TIMESPAN_MAX,
+                                                        address->cinfo.ifa_prefered * USEC_PER_SEC,
+                                                        USEC_PER_SEC);
+
+                log_link_debug(link, "%s address: %s%s%s/%u (valid %s%s, preferred %s%s)",
                                str, strnull(addr), has_peer ? " peer " : "",
                                has_peer ? strnull(peer) : "", address->prefixlen,
-                               valid_str ? "for " : "forever", strempty(valid_str));
+                               valid_str ? "for " : "forever", strempty(valid_str),
```

Comment:
the two strings are being checked, so there's no point in using strempty?

Reply:
Hm? Sorry, I cannot catch what you want to say... Are there any smarter way to prefix `"for "` or just print "forever"?

Reply:
Ignore me, I misread and in my head strempty was after ":" - sorry for the noise